skyportal/handlers/api/observation.py

```diff
@@ -613,3 +616,65 @@ def post(self):
         )
 
         return self.success()
+
+
+class ObservationExternalAPIHandler(BaseHandler):
+    @permissions(['Upload data'])
+    def post(self):
+        """
+        ---
+        description: Retrieve observations from external API
+        tags:
+          - observations
+        requestBody:
+          content:
+            application/json:
+              schema: ObservationExternalAPIHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: ArrayOfExecutedObservations
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+
+        try:
+            data = ObservationExternalAPIHandlerPost.load(data)
+        except ValidationError as e:
+            return self.error(
+                f'Invalid / missing parameters: {e.normalized_messages()}'
+            )
+
+        data["requester_id"] = self.associated_user_object.id
+        data["last_modified_by_id"] = self.associated_user_object.id
+        data['allocation_id'] = int(data['allocation_id'])
+        data['start_date'] = arrow.get(data['start_date'].strip()).datetime
+        data['end_date'] = arrow.get(data['end_date'].strip()).datetime
+
+        allocation = Allocation.get_if_accessible_by(
+            data['allocation_id'], self.current_user, raise_if_none=True
+        )
+        instrument = allocation.instrument
+
+        if instrument.api_classname_obsplan is None:
+            return self.error('Instrument has no remote observation plan API.')
+
+        if not instrument.api_class_obsplan.implements()['retrieve']:
+            return self.error(
+                'Cannot submit executed observation plan requests to this Instrument.'
+            )
+
+        try:
+            instrument.api_class_obsplan.retrieve(
+                allocation, data['start_date'], data['end_date']
+            )
+            self.push_notification(
+                'Observation ingestion in progress. Should be available soon.'
+            )
+        except Exception as e:
+            return self.error(f"Error in querying instrument API: {e}")
```

Comment:
I don't get it: why retrieve at the end of the post?

Reply:
It's a bit of a funny name, but "retrieve" grabs executed observations from the API and commits them. Do you prefer a different name @guynir42 ?

Reply:
I am not sure if a different name is the thing. Maybe add a comment explaining what's happening.

Reply:
added a comment